libs/portals/admin/ids-admin/src/components/forms/AddPermissions/AddPermissions.tsx

```diff
@@ -8,125 +8,167 @@ import {
+  >(new Map())
+  const params = useParams()
+  const [availableScopes, setAvailableScopes] = React.useState<
+    AuthAdminClientAllowedScope[]
+  >([])
+
+  const {
+    selectedEnvironment,
+    setAddedScopes,
+    addedScopes,
+    removedScopes,
+    setRemovedScopes,
+  } = useContext(ClientContext)
+
+  const { loading } = useGetAvailableScopesQuery({
+    fetchPolicy: 'network-only',
+    variables: {
+      input: {
+        tenantId: params['tenant'] ?? '',
+      },
+    },
+    onCompleted: (data) => {
+      const scopes = data.authAdminScopes?.data
+        .map((item) => {
+          return (
+            item.environments.find(
+              (scope) => scope.environment === selectedEnvironment.environment,
+            ) ?? null
+          )
+        })
+        .filter((item) => item !== null)
+
+      setAvailableScopes(scopes as AuthAdminClientAllowedScope[])
+    },
+  })
+  // Get the available scopes for the selected environment including the scopes that have already been deleted
+  const available = availableScopes
+    ?.filter((item) => {
+      return ![
+        ...addedScopes,
+        ...(selectedEnvironment?.allowedScopes ?? []),
+      ].find((added) => added.name === item.name)
+    })
+    .concat(removedScopes)
+
+  // Add the selected scopes to the addedScopes array for
+  const handleAdd = () => {
+    // Combine the previously added scopes and the newly selected scopes
+    const newAddedScopes = [...addedScopes, ...selected.values()]
 
-const mockData: Permission[] = [
-  {
-    label: 'Staða og hreyfingar',
-    id: '@island.is/finance:overview',
-    description:
-      'Skoða stöðu við ríkissjóð og stofnanir, hreyfingar, greiðsluseðla og greiðslukvittanir.',
-    api: 'Island.is APIs',
-    locked: false,
-  },
-  {
-    label: 'Full Access',
-    id: '@island.is/auth/admin:full',
-    description:
-      'Full access to authorization admin something description here',
-    api: 'Island.is APIs',
-    locked: false,
-  },
-  {
-    label: 'Skattskýrslur',
-    id: '@skatturinn.is/skattskyrslur',
-    description:
-      'Full access to authorization admin something description here',
-    api: 'Skatturinn',
-    locked: true,
-  },
-  {
-    label: 'Staða og hreyfingar',
-    id: '@island.is/finance',
-    description:
-      'Skoða stöðu við ríkissjóð og stofnanir, hreyfingar, greiðsluseðla og greiðslukvittanir.',
-    api: 'Island.is APIs',
-    locked: false,
-  },
-  {
-    label: 'Full Access',
-    id: '@island.is/auth/admin',
-    description:
-      'Full access to authorization admin something description here',
-    api: 'Island.is APIs',
-    locked: false,
-  },
-]
+    // Remove the scopes that were added that were also in the removedScopes array
+    setAddedScopes(
+      newAddedScopes.filter(
+        (item) => !removedScopes.some((rem) => rem.name === item.name),
+      ),
+    )
+    // Remove the scopes that were added from the removedScopes array
+    setRemovedScopes(
```

Comment:
This should just be handled in the `onAdd` function of the `Permissions` component.